### PR TITLE
Application default credentials support for GKE provider

### PR DESCRIPTION
**Pull request by maratoid**
Allow for use of [application default credentials](https://developers.google.com/identity/protocols/application-default-credentials) in GKE provider.

Without this patch applied, when running on GKE, talking to various GCP 
services like stackdriver, bigquery, etc. would require manually managing 
.p12 or .json credential keys.

### CODE REVIEW

File: ansible/roles/kraken.provider/kraken.provider.gke/tasks/destroy-cluster.yaml [hunk 1]
```diff
@@ -19,4 +23,6 @@
       --zone {{ cluster.providerConfig.zone.primaryZone }}
       --no-async
       --quiet
-  when: cluster_check.stdout == cluster.name
+  when: cluster_check.stdout == "{{ cluster.name }}"
```
Comment (davidewatson) [inline ansible/roles/kraken.provider/kraken.provider.gke/tasks/destroy-cluster.yaml]: "[You] don’t need to use {{ }} to use variables inside conditionals, as these are already implied."

File: schemas/config/v1/gkeKeyPairConfig.json [hunk 2]
```diff
@@ -13,12 +13,17 @@
...
+      "description": "Default application credentials key file.",
+      "type": "string"
     }
   },
 
   "required": [
     "serviceAccount",
-    "serviceAccountKeyFile"
+    "serviceAccountKeyFile",
+    "defaultApplicationCredentials"
```
Comment (davidewatson) [inline schemas/config/v1/gkeKeyPairConfig.json]: Did this ever work? We can't add required fields without breaking backwards compatibility.

Comment (maratoid) [reply schemas/config/v1/gkeKeyPairConfig.json]: defaultApplicationCredentials should probably be optional. you're right.

File: Documentation/kraken-configs/keypair/gke.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 | Key Name | Required | Type | Description|
 | --- | --- | --- | --- |
 | username | Optional | String | GCE user name for the privatekeyFile/publickeyFile. I.e. 'core' |
-| serviceAccount | Optional | String | Service account name, i.e serviceaccount@project-name.iam.gserviceaccount.com |
-| serviceAccountKeyFile | Optional | String | Service acount key file. |
+| serviceAccount | Required | String | Service account name, i.e serviceaccount@project-name.iam.gserviceaccount.com |
```
Comment (joejulian) [inline Documentation/kraken-configs/keypair/gke.md]: It's my understanding that these are not required (and would be a hindrance) if you have the serviceAccountPasswordFile - pkcs12 certificate based authentication.

**Comment by joejulian** [comment]
This PR is still awaiting requested changes. Will you be able to address these @maratoid?

File: Documentation/kraken-configs/keypair/gke.md (hunk 3 above)
Comment (maratoid) [reply Documentation/kraken-configs/keypair/gke.md]: So, I guess, either one or the other is required. Do we resolve this through documentation?

**Comment by maratoid** [comment]
I'm otherwise occupied at the moment, but I will be able to get back to this.

**Comment by joejulian** [comment]
Until you're ready to get back to this, I'm adding the "do not merge" tag so I can avoid clicking on it every time I look at open PRs. Feel free to remove it when you get back to it.

**Pull request closed without merging**